### ISSUE TITLE
Definition of 'MLKText' must be imported from module 'MLKitTextRecognitionCommon.MLKText' before it is required

### ISSUE THREAD

**Issue by sinshev** [bharat-biradar/Google-Ml-Kit-plugin#117]
When building on iOS I am getting 5 errors:
1) "Declaration of 'MLKTextRecognizedLanguage' must be imported from module 'MLKitTextRecognitionCommon.MLKTextRecognizedLanguage' before it is required"
2) "Definition of 'MLKText' must be imported from module 'MLKitTextRecognitionCommon.MLKText' before it is required"
3) "Definition of 'MLKTextBlock' must be imported from module 'MLKitTextRecognitionCommon.MLKTextBlock' before it is required"
4) "Definition of 'MLKTextLine' must be imported from module 'MLKitTextRecognitionCommon.MLKTextLine' before it is required"
5) "Definition of 'MLKTextElement' must be imported from module 'MLKitTextRecognitionCommon.MLKTextElement' before it is required"

I've tried both 0.6.0 and 0.7.0 versions of the plugin.
`flutter clean` does not help.

**Comment by dustin-graham** [comment]
I am also getting this issue. I'm on an M1 Mac if that helps.

**Comment by minamax1994** [comment]
Same issue here, on flutter v2.2.3 stable, on Mac OS X 10.15.7.

**Comment by dustin-graham** [comment]
@minamax1994 , are you also on a Mac M1 chip?

**Comment by sgm17** [comment]
- Declaration of 'MLKTextRecognizedLanguage' must be imported from module 'MLKitTextRecognitionCommon.MLKTextRecognizedLanguage' before it is required
- Definition of 'MLKText' must be imported from module 'MLKitTextRecognitionCommon.MLKText' before it is required
- Definition of 'MLKTextBlock' must be imported from module 'MLKitTextRecognitionCommon.MLKTextBlock' before it is required
- Definition of 'MLKTextLine' must be imported from module 'MLKitTextRecognitionCommon.MLKTextLine' before it is required
- Definition of 'MLKTextElement' must be imported from module 'MLKitTextRecognitionCommon.MLKTextElement' before it is required

Same issue here, using MacBook Air M1 and google_ml_kit: ^0.7.0

**Comment by canibal1** [comment]
Same issue here, I try flutter clean , delete  `Podfile`, `Podfile.lock`, `Pods/,` `.symlinks` . I try used iOS and `google_ml_kit : ^0.5.0`, `^0.6.0 `and `^0.7.0`. Five days ago (11.08.2021) working good but since last Thursday (12.08.2021) become not working....

**Comment by minamax1994** [comment]
> @minamax1994 , are you also on a Mac M1 chip?

@dustin-graham Not sure, I'm using AppCenter for generating builds.

**Comment by LebranD** [comment]
same issue

**Comment by dustin-graham** [comment]
Ok, I did some digging and experimenting this morning and found some interesting things.

My podfile.lock says that I'm depending on `MLKitTextRecognition (1.4.0-beta1):`

In that version of the dependency, `MLKitTextRecognition.h` looks like this:
```Objective-C
#import "MLKTextRecognizer+Latin.h"
#import "MLKTextRecognizerOptions.h"
```

1.4.0-beta1 is actually Text Recognition V2. In the latest version of V1, 1.3.0, `MLKitTextRecognition.h` looks like this:
```Objective-C
#import "MLKText.h"
#import "MLKTextBlock.h"
#import "MLKTextElement.h"
#import "MLKTextLine.h"
#import "MLKTextRecognizedLanguage.h"
#import "MLKTextRecognizer.h"
```

The code in this repository is written assuming the older, non-beta version of MLKit. So the question is, why are we now getting this incorrect dependency, regardless of what version of this plugin we use. @canibal1 , thanks for trying all those other versions, that was actually the key to figuring this out.

It turns out that the podspec for this repo includes unbounded dependencies on MLKit. So when Google released 1.4.0-beta1 on 8/11/21, all new pod installations pulled in 1.4.0 and broke this plugin.

https://github.com/bharat-biradar/Google-Ml-Kit-plugin/blob/aabc7eb2b98a2db07f3eca40648e28fb35162c9a/ios/google_ml_kit.podspec#L24

I was able to fix this issue locally. I'm going to go make a PR to fix this this morning.

**Comment by canibal1** [comment]
> Ok, I did some digging and experimenting this morning and found some interesting things.
> 
> My podfile.lock says that I'm depending on `MLKitTextRecognition (1.4.0-beta1):`
> 
> In that version of the dependency, `MLKitTextRecognition.h` looks like this:
> 
> ```objc
> #import "MLKTextRecognizer+Latin.h"
> #import "MLKTextRecognizerOptions.h"
> ```
> 
> 1.4.0-beta1 is actually Text Recognition V2. In the latest version of V1, 1.3.0, `MLKitTextRecognition.h` looks like this:
> 
> ```objc
> #import "MLKText.h"
> #import "MLKTextBlock.h"
> #import "MLKTextElement.h"
> #import "MLKTextLine.h"
> #import "MLKTextRecognizedLanguage.h"
> #import "MLKTextRecognizer.h"
> ```
> 
> The code in this repository is written assuming the older, non-beta version of MLKit. So the question is, why are we now getting this incorrect dependency, regardless of what version of this plugin we use. @canibal1 , thanks for trying all those other versions, that was actually the key to figuring this out.
> 
> It turns out that the podspec for this repo includes unbounded dependencies on MLKit. So when Google released 1.4.0-beta1 on 8/11/21, all new pod installations pulled in 1.4.0 and broke this plugin.
> 
> https://github.com/bharat-biradar/Google-Ml-Kit-plugin/blob/aabc7eb2b98a2db07f3eca40648e28fb35162c9a/ios/google_ml_kit.podspec#L24
> 
> I was able to fix this issue locally. I'm going to go make a PR to fix this this morning.

I am very happy that the problem has been resolved. It saved me from a great burden. :)

**Comment by fbernaly** [comment]
I merged this PR: https://github.com/bharat-biradar/Google-Ml-Kit-plugin/pull/118. Does that fix your issue?

**Comment by canibal1** [comment]
> I merged this PR: #118. Does that fix your issue?

Yes, I tried a couple minute ago. It's working very well.
Just add pubspec.yaml file this git adress like that

```
 google_ml_kit:
    git:
      url:https://github.com/dustin-graham/Google-Ml-Kit-plugin 

```
and go ios file then,

`delete Podfile`
`delete Podfile.lock`
`delete Pods/`
`delete .symlinks`
`flutter clean`
`Pub get`
and run project.
Happy build debug and compiles 😄

**Comment by luvishq** [comment]
@canibal1 : flutter run works but building on iOS through xCode  fails , getting the below issue : 

Undefined symbol: _MLKPoseLandmarkTypeRightWrist
Undefined symbol: _MLKPoseLandmarkTypeRightThumb
Undefined symbol: _MLKPoseLandmarkTypeRightKnee

Any clue how to fix this ?

**Comment by dustin-graham** [comment]
@luvishq , I'm not sure about that. I'm not seeing that on my end. Just to make sure we have a good starting point. Make sure you do a flutter clean. In xcode clean the build folder. Delete the pods folder. Then run a pod get, and run again. 

Beyond that I'm not sure. Take a closer look at the error and see where it's happening. Can you provide more information about the error?

**Comment by luvishq** [comment]
@dustin-graham : I tried the above but it failed again, it seems there are some undefined symbols , Please see attached screenshot for the build error I am getting :

<img width="739" alt="Screen Shot 2021-08-22 at 9 48 45 PM" src="https://user-images.githubusercontent.com/70883221/130391433-2253f117-461b-4fc9-985e-d55aaa5886b7.png">

**Comment by nachtmaar** [comment]
@luvishq I had the same problem. A deletion of the Podfile.lock, flutter clean and cleaning derived data folder helped in my case. Of course, also checking out the fix PR as mentioned above by @canibal1.

**Comment by officialFlutterDeveloper** [comment]
Honestly im so so annoyed and im so angry why can't I make this package work ! :(

**Comment by officialFlutterDeveloper** [comment]
Launching lib/main.dart on iPhone 12 in debug mode...
Running pod install...
Running Xcode build...
Xcode build done.                                           12.5s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[50997]: Class AMSupportURLConnectionDelegate is implemented in both /usr/lib/libauthinstall.dylib (0x1fd2dfad8) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x1140882b8). One of the two will be used. Which one is undefined.
    objc[50997]: Class AMSupportURLSession is implemented in both /usr/lib/libauthinstall.dylib (0x1fd2dfb28) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x114088308). One of the two will be used. Which one is undefined.
    ** BUILD FAILED **


Xcode's output:
↳
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:91:26: error: declaration of 'MLKTextRecognizedLanguage' must be imported from module 'MLKitTextRecognitionCommon.MLKTextRecognizedLanguage' before it is required
          languages:(NSArray<MLKTextRecognizedLanguage *> *)languages
                             ^
    In module 'MLKitTextRecognition' imported from /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:3:
    In module 'MLKitTextRecognitionCommon' imported from /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognition/Frameworks/MLKitTextRecognition.framework/Headers/MLKTextRecognizer+Latin.h:4:
    /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognitionCommon/Frameworks/MLKitTextRecognitionCommon.framework/Headers/MLKTextRecognizedLanguage.h:7:12: note: declaration here is not visible
    @interface MLKTextRecognizedLanguage : NSObject
               ^
    /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:28:41: warning: 'textRecognizer' is deprecated: Please use `textRecognizer(options:)` instead. [-Wdeprecated-declarations]
        textRecognizer = [MLKTextRecognizer textRecognizer];
                                            ^
    In module 'MLKitTextRecognition' imported from /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:3:
    /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognition/Frameworks/MLKitTextRecognition.framework/Headers/MLKTextRecognizer+Latin.h:21:5: note: 'textRecognizer' has been explicitly marked deprecated here
        __deprecated_msg("Please use `textRecognizer(options:)` instead.");
        ^
    In module 'UIKit' imported from /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/Target Support Files/google_ml_kit/google_ml_kit-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.5.sdk/usr/include/sys/cdefs.h:200:48: note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:40:42: error: definition of 'MLKText' must be imported from module 'MLKitTextRecognitionCommon.MLKText' before it is required
            textResult[@"text"] = visionText.text;
                                             ^
    In module 'MLKitTextRecognition' imported from /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:3:
    In module 'MLKitTextRecognitionCommon' imported from /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognition/Frameworks/MLKitTextRecognition.framework/Headers/MLKTextRecognizer+Latin.h:4:
    /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognitionCommon/Frameworks/MLKitTextRecognitionCommon.framework/Headers/MLKText.h:9:12: note: definition here is not reachable
    @interface MLKText : NSObject
               ^
    /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:47:33: error: definition of 'MLKTextBlock' must be imported from module 'MLKitTextRecognitionCommon.MLKTextBlock' before it is required
                 cornerPoints:block.cornerPoints
                                    ^
    In module 'MLKitTextRecognition' imported from /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:3:
    In module 'MLKitTextRecognitionCommon' imported from /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognition/Frameworks/MLKitTextRecognition.framework/Headers/MLKTextRecognizer+Latin.h:4:
    /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognitionCommon/Frameworks/MLKitTextRecognitionCommon.framework/Headers/MLKTextBlock.h:11:12: note: definition here is not reachable
    @interface MLKTextBlock : NSObject
               ^
    /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:57:36: error: definition of 'MLKTextLine' must be imported from module 'MLKitTextRecognitionCommon.MLKTextLine' before it is required
                     cornerPoints:line.cornerPoints
                                       ^
    In module 'MLKitTextRecognition' imported from /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:3:
    In module 'MLKitTextRecognitionCommon' imported from /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognition/Frameworks/MLKitTextRecognition.framework/Headers/MLKTextRecognizer+Latin.h:4:
    /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognitionCommon/Frameworks/MLKitTextRecognitionCommon.framework/Headers/MLKTextLine.h:11:12: note: definition here is not reachable
    @interface MLKTextLine : NSObject
               ^
    /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:67:43: error: definition of 'MLKTextElement' must be imported from module 'MLKitTextRecognitionCommon.MLKTextElement' before it is required
                         cornerPoints:element.cornerPoints
                                              ^
    In module 'MLKitTextRecognition' imported from /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/vision/TextRecognizer.m:3:
    In module 'MLKitTextRecognitionCommon' imported from /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognition/Frameworks/MLKitTextRecognition.framework/Headers/MLKTextRecognizer+Latin.h:4:
    /Users/farhanlatif/AndroidStudioProjects/ml_advice_app/ios/Pods/MLKitTextRecognitionCommon/Frameworks/MLKitTextRecognitionCommon.framework/Headers/MLKTextElement.h:13:12: note: definition here is not reachable
    @interface MLKTextElement : NSObject
               ^
    1 warning and 5 errors generated.
    In file included from /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/GoogleMlKitPlugin.m:1:
    /Users/farhanlatif/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/google_ml_kit-0.7.0/ios/Classes/GoogleMlKitPlugin.h:40:22: warning: unused function 'getFlutterError' [-Wunused-function]
    static FlutterError *getFlutterError(NSError *error) {
                         ^
    1 warning generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

Could not build the application for the simulator.
Error launching application on iPhone 12.

**Comment by officialFlutterDeveloper** [comment]
it works on android but when I run here it throws error I have never been able to make ML kit work on IOS  please someone help me I even to padlock and changed the following 

    - MLKitTextRecognition (~> 1.4.0-beta1)

to 

 - MLKitTextRecognition (~> 1.3.0)

nothing works

**Comment by luvishq** [comment]
> @luvishq I had the same problem. A deletion of the Podfile.lock, flutter clean and cleaning derived data folder helped in my case. Of course, also checking out the fix PR as mentioned above by @canibal1.

@dustin-graham @nachtmaar : Still getting the same error :/

**Comment by nachtmaar** [comment]
@luvishq I had this additional error message: "building for iOS Simulator-x86_64 but attempting to link with file built for iOS Simulator-arm64". 

Do you also have a Mac with arm chip ?

if yes, these commands helped:

```shell
rm -r /Users/<your-user>/Library/Developer/Xcode/DerivedData/*
flutter clean
flutter pub get
cd ios
pod install
cd ..
flutter run
```

**Comment by officialFlutterDeveloper** [comment]
still can't make it work im using m1 Mac can anyone help me?

**Comment by officialFlutterDeveloper** [comment]
@dustin-graham any idea how to fix the error?

**Comment by mathrb22** [comment]
> > @luvishq I had the same problem. A deletion of the Podfile.lock, flutter clean and cleaning derived data folder helped in my case. Of course, also checking out the fix PR as mentioned above by @canibal1.
> 
> @dustin-graham @nachtmaar : Still getting the same error :/

Hey @luvishq, I'm also having the same error when trying to build on XCode. I tried deleting the Podfile.lock, flutter clean and cleaning derived data folder but it didn't work either. I also changed the pubspec.yaml plugin reference to the following:

```
 google_ml_kit: 
  git:
    url: git://github.com/bharat-biradar/Google-Ml-Kit-plugin.git
```

I've heard that is probably because of the `1.0.0-beta4` version on the MLKitPoseDetection plugin. Have you found a solution? Thank you in advance.

_flutter doctor:_
```
✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-x64, locale pt-BR)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[!] Xcode - develop for iOS and macOS
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] VS Code (version 1.46.1)
[✓] Connected device (2 available)
```

**Comment by canibal1** [comment]
Hey guys @officialFlutterDeveloper  @mathrb22 @luvishq @,

Please try last version on the ios. It's fixed with last version. But if you still have an issue you can use my forked repo. Because I doesn't change anything. Maybe "github.com/bharat-biradar/Google-Ml-Kit-plugin.git" this github repo changed. I don't know. For my repo:

```
google_ml_kit: 
  git:
    url: git://github.com/canibal1/Google-Ml-Kit-plugin.git
```

you can use it with as I mentioned comment https://github.com/bharat-biradar/Google-Ml-Kit-plugin/issues/117#issuecomment-899836826. Just change with new url.

**Comment by ljmatan** [comment]
Still won't work for me, cleaned derived data, ran `flutter pub cache repair`, etc. 

Also on M1 Mac

**Comment by Aman-Malhotra** [comment]
could it be something specific to M1 ?
did all of the above but doesnt build on M1 chip mac.
@ljmatan @officialFlutterDeveloper  both are facing this on m1 and unable to solve.

**Comment by FedurovKonstantin** [comment]
> luvishq

Hi i had same issue now and what did you do with that error?

**Comment by canibal1** [comment]
Hey guys,

In this case, please analyze the problem by checking the error messages in flutter console and XCode console. 
Because there are more descriptive error messages. XCode can sometimes show exactly where it went wrong.
### If you are using m1, don't forget to do these steps.

Make sure you are using the correct CocoaPods "`sudo arch -x86_64 gem install ffi`"and do the pod install with "`arch -x86_64 pod install`". 
```
cd ios

rm -rf  Podfile.lock
rm -rf   Pods/
rm -rf  .symlinks

flutter clean
flutter pub get

sudo arch -x86_64 gem install ffi
arch -x86_64 pod install
```

then run your code.
If you not use m1 you can use normal way.

**Comment by fbernaly** [comment]
We released a new version of [google_ml_kit](https://github.com/bharat-biradar/Google-Ml-Kit-plugin/tree/master/packages/google_ml_kit) in which we have split all the plugins in multiple plugins to avoid removing the models for API you are not using, go to our readme and start using the new plugin you need.

Start using: 

https://github.com/bharat-biradar/Google-Ml-Kit-plugin/tree/master/packages/google_mlkit_text_recognition

More details here:

https://github.com/bharat-biradar/Google-Ml-Kit-plugin/tree/master/packages/google_ml_kit

**Comment by ashishktrn** [comment]
> When building on iOS I am getting 5 errors:
> 
> 1. "Declaration of 'MLKTextRecognizedLanguage' must be imported from module 'MLKitTextRecognitionCommon.MLKTextRecognizedLanguage' before it is required"
> 2. "Definition of 'MLKText' must be imported from module 'MLKitTextRecognitionCommon.MLKText' before it is required"
> 3. "Definition of 'MLKTextBlock' must be imported from module 'MLKitTextRecognitionCommon.MLKTextBlock' before it is required"
> 4. "Definition of 'MLKTextLine' must be imported from module 'MLKitTextRecognitionCommon.MLKTextLine' before it is required"
> 5. "Definition of 'MLKTextElement' must be imported from module 'MLKitTextRecognitionCommon.MLKTextElement' before it is required"
> 
> I've tried both 0.6.0 and 0.7.0 versions of the plugin. `flutter clean` does not help.

have you found any solution? I'm facing the same problem please help me

**Comment by ashishktrn** [comment]
> > Ok, I did some digging and experimenting this morning and found some interesting things.
> > My podfile.lock says that I'm depending on `MLKitTextRecognition (1.4.0-beta1):`
> > In that version of the dependency, `MLKitTextRecognition.h` looks like this:
> > ```objc
> > #import "MLKTextRecognizer+Latin.h"
> > #import "MLKTextRecognizerOptions.h"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 1.4.0-beta1 is actually Text Recognition V2. In the latest version of V1, 1.3.0, `MLKitTextRecognition.h` looks like this:
> > ```objc
> > #import "MLKText.h"
> > #import "MLKTextBlock.h"
> > #import "MLKTextElement.h"
> > #import "MLKTextLine.h"
> > #import "MLKTextRecognizedLanguage.h"
> > #import "MLKTextRecognizer.h"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The code in this repository is written assuming the older, non-beta version of MLKit. So the question is, why are we now getting this incorrect dependency, regardless of what version of this plugin we use. @canibal1 , thanks for trying all those other versions, that was actually the key to figuring this out.
> > It turns out that the podspec for this repo includes unbounded dependencies on MLKit. So when Google released 1.4.0-beta1 on 8/11/21, all new pod installations pulled in 1.4.0 and broke this plugin.
> > https://github.com/bharat-biradar/Google-Ml-Kit-plugin/blob/aabc7eb2b98a2db07f3eca40648e28fb35162c9a/ios/google_ml_kit.podspec#L24
> > 
> > I was able to fix this issue locally. I'm going to go make a PR to fix this this morning.
> 
> I am very happy that the problem has been resolved. It saved me from a great burden. :)

please help me how you fixed this??

**Comment by suryadwipayana1001** [comment]
> When building on iOS I am getting 5 errors:
> 
> 1. "Declaration of 'MLKTextRecognizedLanguage' must be imported from module 'MLKitTextRecognitionCommon.MLKTextRecognizedLanguage' before it is required"
> 2. "Definition of 'MLKText' must be imported from module 'MLKitTextRecognitionCommon.MLKText' before it is required"
> 3. "Definition of 'MLKTextBlock' must be imported from module 'MLKitTextRecognitionCommon.MLKTextBlock' before it is required"
> 4. "Definition of 'MLKTextLine' must be imported from module 'MLKitTextRecognitionCommon.MLKTextLine' before it is required"
> 5. "Definition of 'MLKTextElement' must be imported from module 'MLKitTextRecognitionCommon.MLKTextElement' before it is required"
> 
> I've tried both 0.6.0 and 0.7.0 versions of the plugin. `flutter clean` does not help.

just add 
@import MLKitTextRecognition;
@import MLKitTextRecognitionCommon;
@import MLKitVision;
then pod install. this work for me

**Comment by tchex** [comment]
> > When building on iOS I am getting 5 errors:
> > 
> > 1. "Declaration of 'MLKTextRecognizedLanguage' must be imported from module 'MLKitTextRecognitionCommon.MLKTextRecognizedLanguage' before it is required"
> > 2. "Definition of 'MLKText' must be imported from module 'MLKitTextRecognitionCommon.MLKText' before it is required"
> > 3. "Definition of 'MLKTextBlock' must be imported from module 'MLKitTextRecognitionCommon.MLKTextBlock' before it is required"
> > 4. "Definition of 'MLKTextLine' must be imported from module 'MLKitTextRecognitionCommon.MLKTextLine' before it is required"
> > 5. "Definition of 'MLKTextElement' must be imported from module 'MLKitTextRecognitionCommon.MLKTextElement' before it is required"
> > 
> > I've tried both 0.6.0 and 0.7.0 versions of the plugin. `flutter clean` does not help.
> 
> just add @import MLKitTextRecognition; @import MLKitTextRecognitionCommon; @import MLKitVision; then pod install. this work for me

Hello @suryadwipayana1001 
Where / how do you add these imports?
Thanks

**Comment by tchex** [comment]
> @import MLKitTextRecognition;
> @import MLKitTextRecognitionCommon;
> @import MLKitVision;

Thanks a lot, I don't have this error anymore!
Now facing "CocoaPods could not find compatible versions for pod "GoogleMLKit/TextRecognition"".. but it's another story.

**Comment by lapagydem** [comment]
> @import MLKitTextRecognition;
> @import MLKitTextRecognitionCommon;
> @import MLKitVision;

This resolved my issue, Thanks
